UltimateHelperEx/CryptographyHelper.cs

```
@@ -16,129 +16,65 @@ namespace DataJuggler.Core.UltimateHelper
+                // Remember to dispose of types that implement IDisposable - this old code had lots of memory leaks.
+                using (var aes = AesManaged.Create())
+                using (var pbkdf2 = new Rfc2898DeriveBytes(password, _pepper, 32767)) // MD5 is insecure as KDF, we use PBKDF2 instead.
+                using (var rng = new RNGCryptoServiceProvider())
+                {
+                    var key = pbkdf2.GetBytes(32); // Let's use AES-256.
+                    var iv = new byte[16];
+                    rng.GetBytes(iv); // We always create a new, random IV for each operation.
 
-                        // Electronic Code Book Cipher
-                        des.Mode = CipherMode.ECB; //CBC, CFB
+                    var plaintext = Encoding.UTF8.GetBytes(stringToEncrypt); // We use UTF8 - there is literally no reason to use ASCII in 2019.
 
-                        // Set Buffer To StringToEncrypt
-                        buff = ASCIIEncoding.ASCII.GetBytes(stringToEncrypt);
+                    aes.Mode = CipherMode.CBC;
+                    aes.Padding = PaddingMode.PKCS7;
+                    aes.Key = key;
+                    aes.IV = iv;
 
-                        // Get Encrypted String
-                        encryptedString = Convert.ToBase64String(des.CreateEncryptor().TransformFinalBlock(buff, 0, buff.Length));
+                    using (var aesTransformer = aes.CreateEncryptor())
+                    {
+                        var ciphertext = aesTransformer.TransformFinalBlock(plaintext, 0, plaintext.Length);
+                        var ivAndCiphertext = new byte[iv.Length + ciphertext.Length];
+                        Array.Copy(iv, 0, ivAndCiphertext, 0, iv.Length);
+                        Array.Copy(ciphertext, 0, ivAndCiphertext, iv.Length, ciphertext.Length);
+                        return Convert.ToBase64String(ivAndCiphertext);
                     }
                 }
-                catch
```

This catch hides all exceptions - bad design.  Fixed to remove this and add null checks to input arguments ✔️